### ISSUE TITLE
Angular Typescript basePath variable doesn't include host and schemes properties

### ISSUE THREAD

**Issue by Aitchy13** [swagger-api/swagger-codegen#1181]
The `host` and `schemes` properties aren't prepended to the `basePath` property within an API class.

For example:

**swagger.yaml**
```
host: "localhost:9000"
schemes:
  - http
basePath: "/1.0"
```

**Output**
```
module API.Client {
    'use strict';

        export class ClientApi {
            private basePath = '/1.0';

...
```

The basePath only outputs the `basePath` property, e.g. `'/1.0'`.

**Link to Angular Typescript mustache template**
https://github.com/swagger-api/swagger-codegen/blob/master/modules/swagger-codegen/src/main/resources/TypeScript-Angular/api.mustache

**Comment by wing328** [comment]
@Aitchy13 thanks for reporting the issue. I agree that ideally the base path should contains the schemes, host, basePath as defined in the Swagger spec.

As a workaround, please pass the full base path to the Api class while we try to address the issue, e.g.
```
var petApi = new api.PetApi('http://petstore.swagger.io');
```

**Comment by wing328** [comment]
@Aitchy13 fixed in #1196 

You can review the sample output for Petstore API here: https://github.com/wing328/swagger-codegen/blob/typescript_basepath/samples/client/petstore/typescript-angular/API/Client/PetApi.ts
